File: imperium-common/src/main/java/com/xpdustry/imperium/common/string/MissingStringRequirementException.java
package com.xpdustry.imperium.common.string;

import java.util.List;

public final class MissingStringRequirementException extends RuntimeException {

    private final List<StringRequirement> missing;

    public MissingStringRequirementException(final String message, final List<StringRequirement> missing) {
        super(message);
        this.missing = List.copyOf(missing);
    }

    public List<StringRequirement> missing() {
        return missing;
    }
}

File: imperium-common/src/main/kotlin/com/xpdustry/imperium/common/string/MissingStringRequirementException.kt
// SPDX-License-Identifier: GPL-3.0-only
package com.xpdustry.imperium.common.string

class MissingStringRequirementException(message: String, missing: List<StringRequirement>) : RuntimeException(message) {
    val missing: List<StringRequirement> = missing.toList()
}
